_posts/2020-05-26-full-text-search.md

```diff
@@ -0,0 +1,74 @@
+
+<figure>
+  <img src="/images/full-text-search/start_search.png" alt="Initial lesson home page, showing the start search button." title="Initial lesson home page, showing the start search button."/>
+  <figcaption>Initial lesson home page. Click start searching to enter search queries.</figcaption>
+</figure>
+
+<br/>
+
+You'll now see a search bar and button. You can enter your search terms and get a list of the relevant  lessons, with the search terms highlighted.
+
+<figure>
+  <img src="/images/full-text-search/search_results.png" alt="Search results displaying highlighted search terms." title="Search results displaying highlighted search terms."/>
+  <figcaption>Search results for <em>Twitter</em> and <em>Network</em></figcaption>
+</figure>
+
+<br/>
+
+The results are ranked by relevance and you can also filter them using our existing buttons.
+
+<figure>
+  <img src="/images/full-text-search/search_filter.png" alt="Search results displaying highlighted search terms with selected filters." title="Search results displaying highlighted search terms with selected filters."/>
+  <figcaption>Search results for <em>Twitter</em> and <em>Network</em> with topic <em>Python</em></figcaption>
+</figure>
+
+<br/>
+If you want more information about searching, you can click the information button to get more details about how to use this feature.
+
+<img src="/images/full-text-search/search_info.png" alt="Search info section, displaying additional details on how to search." title="Search info section, displaying additional details on how to search."/>
+
+## How does the search work?
+
+Behind the scenes, this search feature uses [LunrJS](https://lunrjs.com), a software package for enabling full-text search on static sites.
+<div style="display: flex;">
+  <figure>
+    <img src="/images/full-text-search/inverted_index.jpeg" alt="Search info section, displaying additional details on how to search." title="Search info section, displaying additional details on how to search."/>
+    <figcaption>Inverted index diagram</figcaption>
+  </figure>
+  <figure>
+    <img src="/images/full-text-search/book_index.jpg" alt="Search info section, displaying additional details on how to search." title="Search info section, displaying additional details on how to search."/>
+    <figcaption>Book index (<a href="https://commons.wikimedia.org/wiki/File:Book_of_Knowledge_1919_Vol_20,_General_Index_Start.jpg">from Wikipedia entry on <em>Book of Knowledge</em></a>)</figcaption>
+  </figure>
+</div>
+
+Lunr builds an *inverted index* of all our lessons, which is essentially the same as an index at the back of a book. So each time you enter a search term, Lunr looks for term, finds all the lessons that is in, and then returns the lessons based on relevance of the term (which is calculated using an information retrieval algorithm called Okapi BM25 [https://en.wikipedia.org/wiki/Okapi_BM25](https://en.wikipedia.org/wiki/Okapi_BM25)).
```

Comment:
Check this and other links that appear later in the post - you haven't got the markdown linking quite right - it's `[text here](url)`

Reply:
haha oh jeez 😅 ! Thanks for catching that.... I swear I've used markdown before 🙈